Frame.cc

```diff
@@ -1234,17 +1236,30 @@ bool Frame::GetCursorSpectralData(
             casacore::IPosition count(sub_image_shape);
             float progress = 0.0;
             // get cursor's x-y coordinate from subimage
-            CursorXy cursor_xy;
-            GetSubImageXy(sub_image, cursor_xy);
+            CursorXy subimage_cursor;
+            GetSubImageXy(sub_image, subimage_cursor);
             // get spectral profile for the cursor
             auto t_partial_profile_start = std::chrono::high_resolution_clock::now();
             while (start(_spectral_axis) < profile_size) {
                 // start the timer
                 auto t_start = std::chrono::high_resolution_clock::now();
-                // check if frontend's requirements changed
-                if (Interrupt(cursor_xy)) {
-                    return false;
+                // check if region point changed from subimage point
+                if ((region_id == CURSOR_REGION_ID) && (Interrupt(_cursor_xy, subimage_cursor))) {
+                    return false; // cursor moved
                 }
+                if (region_id > CURSOR_REGION_ID) {
```

Comment:
It seems the `region_id` is always equal to 0. Because we have already set to execute the function `GetPointSpectralData()` only when it is a point region (on line 923 `if (region->IsPoint()) {..}`).

Reply:
A point region can be a cursor (region id 0) or a fixed point region (SET_REGION with type Point and id -1 which gets set to the next available region id).  Point regions are being implemented for release 1.2.

Reply:
I see. So @kswang1029's problem is due to the lack of interruption condition that users switch between different regions or cursor. I will do it and send another PR. You can merge this PR first 
since it is irrelevant to the subject.

Reply:
@kswang1029's problem is that the cursor spectral profile doesn't know that its requirements have been changed (to profiles=[], which should cancel).  In addition the cursor/point position, the number of profiles should be checked, and the profile coordinate (stokes) should be checked (for example if user changes I -> Q you do not need to send stokes I any more).  For non-point regions, the region position, stokes, and requested stats should be checked.